funsor/delta.py

```diff
@@ -129,14 +129,14 @@ def eager_add(op, lhs, rhs):
 
 @eager.register(Independent, Delta, str, str)
 def eager_independent(delta, reals_var, bint_var):
-    if delta.name == reals_var:
+    if delta.name == reals_var or delta.name.startswith(reals_var + "__BOUND"):
```

Comment:
Can you explain under what circumstances `delta` should substitute into a different variable? This seems fishy on first look.

Reply:
Yes, there are some odd-looking changes here because the semantics of `Independent` are complicated (although I would argue that the changes in this PR clarify the type of `Independent`).  `Independent` binds `reals_var` in its input expression, leading to alpha-renaming, and introduces a fresh variable with the same name but a different type.  This condition is checking if `delta` contains the *bound* variable originally named `reals_var`, which may or may not have undergone alpha-conversion.

Reply:
Hmm do you think it would be less error-prone to change `Independent` to require two user-provided names, one for the bound and one for the fresh? If you think it's fine as-is could you just add a little comment noting what you explained to me?